### PR TITLE
Add keyword argument to `nx.ancestors` & `nx.descendants`

### PR DESCRIPTION
**Pull request by dtekinoglu**
PR for #5801. I added keyword argument `include_source` to `nx.ancestors` and `nx.descendants` methods. If it is set to True, the list returned by the method will include the source node itself as its ancestor/descendant.

### CODE REVIEW

**Review by jarrodmillman — APPROVED** [review]

File: networkx/algorithms/dag.py [hunk 1]
```diff
@@ -35,13 +35,16 @@
...
+def descendants(G, source, include_source=False):
     """Returns all nodes reachable from `source` in `G`.
 
     Parameters
     ----------
     G : NetworkX Graph
     source : node in `G`
+    include_source: bool, optional (default=False)
+        If False, the source is not included as its own descendant. If True, the source node as well
+        is returned as one of its descendants.
```
Comment (dschult) [inline networkx/algorithms/dag.py:47]:
```suggestion
        If False, the source is not included as its own descendant. 
        If True, the source node as well is returned as one of its descendants.
```

File: networkx/algorithms/dag.py [hunk 2]
```diff
@@ -59,20 +62,29 @@ def descendants(G, source):
...
+def ancestors(G, source, include_source=False):
     """Returns all nodes having a path to `source` in `G`.
 
     Parameters
     ----------
     G : NetworkX Graph
     source : node in `G`
+    include_source: bool, optional (default=False)
+        If False, the source is not included as its own descendant. If True, the source node as well
+        is returned as one of its descendants.
```
Comment (dschult) [inline networkx/algorithms/dag.py:87]:
```suggestion
        If False, the source is not included as its own descendant. 
        If True, the source node as well is returned as one of its descendants.
```

**Review by dschult — COMMENTED** [review]
Here's two spacing nits for the doc_string.

**Comment by rossbar** [comment]
I think at the last meeting we discussed updating the docstring to note that these functions don't include `source` rather than adding a new keyword argument (see [this comment](https://github.com/networkx/networkx/issues/5801#issuecomment-1161598494) for a longer version of my take).

If others prefer the new kwarg instead that's find with me (I don't have a strong preference) but IMO we should make a clear decision somewhere!

**Comment by dtekinoglu** [comment]
I have pushed another commit that reverts changes for keyword argument and adds an example to documentation. Based on the final decision, I can revert this commit as well :) @rossbar

File: networkx/algorithms/dag.py [hunk 3]
```diff
@@ -56,8 +56,13 @@ def descendants(G, source):
     Examples
     --------
     >>> DG = nx.path_graph(5, create_using=nx.DiGraph)
-    >>> sorted(list(nx.descendants(DG, 2)))
-    [3, 4]
+    >>> descendants = nx.descendants(DG, 2)
+    >>> descendants
+    {3, 4}
```
Comment (rossbar) [inline networkx/algorithms/dag.py:61]: We'll need to keep the `sorted` as sets are not ordered, so giving the set output directly might result in intermittent doctest failures. See [the doctest docs on this subject](https://docs.python.org/3/library/doctest.html#warnings) for more info.

```suggestion
    >>> descendants = nx.descendants(DG, 2)
    >>> descendants
    {3, 4}
```
```suggestion
    >>> sorted(nx.descendants(DG, 2))
    [3, 4]
```

File: networkx/algorithms/dag.py [hunk 4]
```diff
@@ -56,8 +56,13 @@ def descendants(G, source):
...
+
+    A node is not returned as one of its own descendantsgit . If you prefer to include the node itself as well, you can perform a manual union operation as follows:
```
Comment (rossbar) [inline networkx/algorithms/dag.py:63]: It looks like `git` snuck itself in here :). A minor wording suggestion (feel free to ignore):

```suggestion
    The `source` node is not a descendant of itself, but can be included manually:
```

File: networkx/algorithms/dag.py [hunk 5]
```diff
@@ -56,8 +56,13 @@ def descendants(G, source):
...
+
+    >>> descendants = descendants.union({2})
```
Comment (rossbar) [inline networkx/algorithms/dag.py:65]: Same deal here with the `sorted`. I also personally like using the binary operators for sets, but if you prefer the explicit method names feel free to ignore!

```suggestion
    >>> descendants = descendants.union({2})
```
```suggestion
    >>> sorted(nx.descendants(DG, 2) | {2})
    [2, 3, 4]
```

**Review by rossbar — COMMENTED** [review]
Thanks @dtekinoglu , I like the doc example updates! I've left a few suggestions for minor improvements but nothing critical. Note that any suggestions you agree with should also be applied to the `ancestors` docstring.

**Review by dschult — APPROVED** [review]

**Review by rossbar — APPROVED** [review]
A very nice update, thanks @dtekinoglu !

**Pull request merged**